sw/device/rom_exts/rom_ext_manifest_parser.c

```diff
@@ -0,0 +1,117 @@
+  return mmio_region_read32(base_addr, ROM_EXT_IMAGE_LENGTH_OFFSET);
+}
+
+uint32_t rom_ext_version_get(mmio_region_t base_addr) {
+  return mmio_region_read32(base_addr, ROM_EXT_IMAGE_VERSION_OFFSET);
+}
+
+uint64_t rom_ext_timestamp_get(mmio_region_t base_addr) {
+  uint32_t timestamp_low =
+      mmio_region_read32(base_addr, ROM_EXT_IMAGE_TIMESTAMP_OFFSET);
+
+  ptrdiff_t offset_high = ROM_EXT_IMAGE_TIMESTAMP_OFFSET + sizeof(uint32_t);
+  uint32_t timestamp_high = mmio_region_read32(base_addr, offset_high);
+
+  return ((uint64_t)timestamp_high << 32) | timestamp_low;
+}
+
+uint32_t rom_ext_algorithm_id_get(mmio_region_t base_addr) {
+  return mmio_region_read32(base_addr,
+                            ROM_EXT_SIGNATURE_ALGORITHM_IDENTIFIER_OFFSET);
+}
+
+uint32_t rom_ext_exponent_get(mmio_region_t base_addr) {
+  return mmio_region_read32(base_addr, ROM_EXT_EXPONENT_OFFSET);
+}
+
+uint32_t rom_ext_usage_constraints_get(mmio_region_t base_addr) {
+  return mmio_region_read32(base_addr, ROM_EXT_USAGE_CONSTRAINTS_OFFSET);
+}
+
+// TODO - `rom_ext_lockdown_info_t` probably should be internal
+//        and parse the information further on before returning.
+bool rom_ext_peripheral_lockdown_info_get(mmio_region_t base_addr,
+                                          rom_ext_lockdown_info_t *dst) {
+  if (dst == NULL) {
+    return false;
+  }
+
+  memcpy(&dst->data[0], (const void *)base_addr.base,
+         ROM_EXT_PERIPHERAL_LOCKDOWN_INFO_SIZE_BYTES);
+
+  return true;
+}
+
+bool rom_ext_public_key_get(mmio_region_t base_addr,
+                            rom_ext_public_key_t *dst) {
+  if (dst == NULL) {
+    return false;
+  }
+
+  memcpy(&dst->data[0], (const void *)base_addr.base,
+         ROM_EXT_SIGNATURE_PUBLIC_KEY_SIZE_BYTES);
+
+  return true;
+}
+
+rom_ext_extension_t rom_ext_extension0_get(mmio_region_t base_addr) {
+  return (rom_ext_extension_t){
+      .offset = mmio_region_read32(base_addr, ROM_EXT_EXTENSION0_OFFSET_OFFSET),
+      .checksum =
+          mmio_region_read32(base_addr, ROM_EXT_EXTENSION0_CHECKSUM_OFFSET),
+  };
+}
+
+rom_ext_extension_t rom_ext_extension1_get(mmio_region_t base_addr) {
+  return (rom_ext_extension_t){
+      .offset = mmio_region_read32(base_addr, ROM_EXT_EXTENSION1_OFFSET_OFFSET),
+      .checksum =
+          mmio_region_read32(base_addr, ROM_EXT_EXTENSION1_CHECKSUM_OFFSET),
+  };
+}
+
+rom_ext_extension_t rom_ext_extension2_get(mmio_region_t base_addr) {
+  return (rom_ext_extension_t){
+      .offset = mmio_region_read32(base_addr, ROM_EXT_EXTENSION2_OFFSET_OFFSET),
+      .checksum =
+          mmio_region_read32(base_addr, ROM_EXT_EXTENSION2_CHECKSUM_OFFSET),
+  };
+}
+
+rom_ext_extension_t rom_ext_extension3_get(mmio_region_t base_addr) {
+  return (rom_ext_extension_t){
+      .offset = mmio_region_read32(base_addr, ROM_EXT_EXTENSION3_OFFSET_OFFSET),
+      .checksum =
+          mmio_region_read32(base_addr, ROM_EXT_EXTENSION3_CHECKSUM_OFFSET),
+  };
+}
```

Comment:
The struct here is good, but let's unify these into a function that takes the extension index, rather than has it in the function name.

Reply:
Sure, however then this function will return `bool` that will result in another if statement in the caller, I was aiming to reduce these where possible (I assume `switch` `default` case the same way that we do in DIFs (error)).

Not a big deal really, so I'll do that.

Reply:
Done.